lib/ca_test.go

```diff
@@ -217,7 +218,7 @@ func testValidMatchingKeys(cert *x509.Certificate, t *testing.T) {
 		t.Error("Should have failed, public key and private key do not match")
 	}
 
-	err = validateMatchingKeys(cert, string(0))
+	err = validateMatchingKeys(cert, strconv.Itoa(0))
```

Comment:
Just a comment: Using Itoa to render the number `0` as the string `"0"` could be done by simply replacing it with the literal `"0"`.

It also looks like most of these changes are related to trying to read or create files in bad paths.

Reply:
>Just a comment: Using Itoa to render the number 0 as the string "0" could be done by simply replacing it with the literal "0"

True enough

>It also looks like most of these changes are related to trying to read or create files in bad paths. I can't tell if the author was initially trying to introduce null terminators into the strings (rune(0) or "\x00") or just come up with a bogus value.

seems like both depending on the test

Reply:
I went with bogus strings to make the intention clearer.